src/cpp/rtps/resources/ResourceEvent.cpp

```diff
@@ -196,11 +196,11 @@ void ResourceEvent::run_io_service()
 void ResourceEvent::init_thread()
 {
     thread_ = std::thread(&ResourceEvent::run_io_service, this);
-    std::promise<bool> ready;
-    std::future<bool> ready_fut = ready.get_future();
+    std::promise<void> ready;
```

Comment:
Ninja change!

Reply:
What ninja change? I don't see any ninja change. It should be a good ninja.